js-core/homeworks/homework-15/src/main.js

```diff
@@ -0,0 +1,386 @@
+            
+            user.forEach(userInfo => {
+                const name = this.newElem('td');
+                const lastName = this.newElem('td');
+                const email = this.newElem('td');
+                
+                this.setTextContent(name, userInfo.name);
+                this.setTextContent(lastName, userInfo.lastName);
+                this.setTextContent(email, userInfo.email);
+                
+                tr.appendChild(name);
+                tr.appendChild(lastName);
+                tr.appendChild(email);
+            });
+            tbody.appendChild(tr);
+        });
+        
+        /*Append these element to table*/
+        this.table.appendChild(thead);
+        this.table.appendChild(tbody);
+    },
+    
+    createMainContent() {
+        this.main = this.newElem('main');
+        const conteiner = this.newElem('div');
+        this.setClass(conteiner, 'container');
+        
+        /*There is implementation of search form*/
+        const searchForm = this.newElem('form');
+        this.setClass(searchForm, 'form-inline search-form');
+        searchForm.appendChild(this.fillSearchForm()); // fillSearchForm is above
+        
+        /*There is implementation of contact list (table)*/
+        this.table = this.newElem('table');
+        this.setClass(this.table, 'table table-hover contacts');
+        this.fillTableContactList();
+        
+        conteiner.appendChild(searchForm);
+        conteiner.appendChild(this.table);
+        this.main.appendChild(conteiner);
+    },
+    
+    fillFooter() {
+        const nav = this.newElem('nav');
+        this.setClass(nav, 'main-nav');
+        
+        footerTabs.forEach((tab, index) => {
+            const link = this.newElem('a');
+            const icon = this.newElem('span');
+            const tabContent = this.newElem('span');
+            
+            this.setClass(link, 'tab');
+            this.setClass(icon, `glyphicon glyphicon-${idOfIcons[index]}`);
+            this.setClass(tabContent, 'tab-text');
+            
+            this.setHrefForLink(link, '#');
+            this.setAriaHidden(icon, true);
+            this.setTextContent(tabContent, tab);
+            
+            link.appendChild(icon);
+            link.appendChild(tabContent);
+            nav.appendChild(link);
+        });
+        
+        /*Make first child active*/
+        
+        return nav;
+    },
+    
+    createFooter() {
+        this.footer = this.newElem('footer');
+        const container = this.newElem('div');
+        
+        this.setClass(this.footer, 'footer');
+        this.setClass(container, 'container bottom-radius');
+        
+        container.appendChild(this.fillFooter());
+        this.footer.appendChild(container);
+    },
+    
+    render() {
+        this.createHeader();
+        this.createMainContent();
+        this.createFooter();
+        
+        document.body.appendChild(this.header);
```

Comment:
it will be better if you do not create a header inside such methods but return them.

It's not obvious if `this.main` mutating during another creator the same idea behind `this.header` and `this.footer`